x-pack/plugins/infra/public/components/asset_details/hooks/use_process_list.ts

```diff
@@ -54,7 +54,10 @@ export function useProcessList(
       sortBy: parsedSortBy,
       searchFilter,
     }),
-    decodeResponse
+    decodeResponse,
+    undefined,
+    undefined,
+    true
```

Comment:
Yeah, I believe this slowness is largely due to the environment being too slow.

Not having an AbortController in place can lead to equally bad consequences, possibly even worse, as it opens the door to multiple unnecessary requests being queued up for later processing by the server.

In the video, we can observe that even the KPI charts didn't load, and they don't have the abort controller mechanism. What's happening here is that new requests are being initiated, marking them as 'loading', while previous ones haven't received responses yet.

I managed to reproduce this by setting the date range to 'Last 7 days', and I faced the same issue.

[Link to the reproduction](https://github.com/elastic/kibana/assets/2767137/32990307-e5ea-4c3a-b3c1-e120d8e76a65)

Aside from the optimizations introduced in this PR, I'm struggling to come up with a straightforward solution to prevent new requests from piling on top of unfinished ones. It may boil down to adjusting the refresh interval and the date picker range.

@roshan-elastic, how weird would it be to include a message below the date picker (when auto-refresh is enabled) advising the user to tweak the auto-refresh interval if the flyout appears stuck in an endless loading state?

Reply:
> @roshan-elastic, how weird would it be to include a message below the date picker (when auto-refresh is enabled) advising the user to tweak the auto-refresh interval if the flyout appears stuck in an endless loading state?

That's a great idea @crespocarlos. Is there a way to detect when it's stuck and only show it then?

I have a question - was this a problem with the existing inventory auto-refresh? If so, I think we can leave it as it is unless it's really easy to apply helpful user-messaging.

Reply:
>  Is there a way to detect when it's stuck and only show it then?

@roshan-elastic, it would be too complicated to detect all loading indicators on a tab.

> I have a question - was this a problem with the existing inventory auto-refresh? If so, I think we can leave it as it is unless it's really easy to apply helpful user-messaging

https://demo.elastic.co/app/metrics/inventory. The previous version was lighter than the asset details flyout - fewer charts, no KPIs, fewer concurrent requests queueing up

Reply:
That's true that we don't want to overload the server with all requests, aborting them should be fine. I think maybe it's the env that is slowing it down. 
Adding a message to tell the user is also an option - it can be more generic if we have a loading state we can show it ( similar to the metadata explanation loading indicator logic - only on the first load when we don't have data). As it will be hard to detect all loading states we can use metadata loading maybe 🤔 This can be also discussed and added in a separate issue, it doesn't need to be part of the PR.

Reply:
> https://demo.elastic.co/app/metrics/inventory. The previous version was lighter than the asset details flyout - fewer charts, no KPIs, fewer concurrent requests queueing up

@crespocarlos  On the general inventory view - would the same thing potentially happen? i.e. could you open inventory and it could take too long for the waffle to show because of auto-refresh?

I'm thinking this could be the same problem but in a different context (in which case, we perhaps don't need to deal with it for the moment)?

> Adding a message to tell the user is also an option - it can be more generic if we have a loading state we can show it ( similar to the metadata explanation loading indicator logic - only on the first load when we don't have data). As it will be hard to detect all loading states we can use metadata loading maybe 🤔 This can be also discussed and added in a separate issue, it doesn't need to be part of the PR.

@jennypavlova - are you thinking that the generic message would be part of this PR or a separate issue?

Reply:
@roshan-elastic 

>  On the general inventory view - would the same thing potentially happen? i.e. could you open inventory and it could take too long for the waffle to show because of auto-refresh?

> I'm thinking this could be the same problem but in a different context (in which case, we perhaps don't need to deal with it for the moment)?

Probably not. I ran a test with a really slow network, and the charts loaded up just fine.

Regarding the old inventory's flyout, it calls a single API (Snapshot API) for the charts. It doesn't interrupt any ongoing requests during refresh cycles. So, if a request was sent 30 seconds ago, it'll eventually get a response, and the flyout will use that data even if there are more recent requests pending

Now, for the charts in the Asset Details: Lens uses an endpoint that makes asynchronous requests to Elasticsearch. That's why we see a bunch of requests to `bsearch` here:

<img width="1242" alt="image" src="https://github.com/elastic/kibana/assets/2767137/c9ca9f99-6fca-4afb-bee0-f5022d4c15dd">

Whenever we hit `bsearch` with a new request, it hands over a unique id. Since an async operation is happening on the Elasticsearch side, Lens uses this id to continuously ask `bsearch` whether the data is good to go.

https://github.com/elastic/kibana/assets/2767137/1cc4541b-d794-4d3b-a941-482cff5797b2

When the auto-refresh kicks in, we tell Lens to make a fresh request with the updated date range. That gets `bsearch` to hand out a whole new unique id. As a result, Lens forgets all about the previous calls and starts checking with 'bsearch' for this new id. That's why we end up in this never-ending loading loop.

cc @jennypavlova